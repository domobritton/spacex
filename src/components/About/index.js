import ScrollAnim from 'rc-scroll-anim'
import React, { Component } from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

import TweenOne from 'rc-tween-one'
import styled from 'styled-components'

const ScrollOverPack = ScrollAnim.OverPack
ScrollAnim.scrollScreen.init({ loop: true })

export default class About extends Component {
    constructor() {
        super()

        this.state = {
            data: [],
        }
  }

  componentDidMount() {
    fetch('https://api.spacexdata.com/v2/info')
      .then(res => res.json())
      .then(data => this.setState({ data: data }))
      .catch(err => console.error(err));
  }

    render() {
        const { data } = this.state 
        return (
            <Page>
              <Image />
              <Scroll
                playScale={1}
                id="page4">
                <Tween
                  animation={{ y: 0, opacity: 1 }}
                  key="7">
                  <Content>    
                      <Tabs>
                        <Folder>
                            <StyledTab>About</StyledTab>
                            <StyledTab>Stats</StyledTab>
                        </Folder>
                    
                        <TabPanel>
                            <Item1>{data.summary}</Item1>
                        </TabPanel>
                        <TabPanel>
                            <Title>Company Info</Title>
                           {data ? 
                           <List>
                                <Item>Founded: {data.founded}</Item>
                                <Item>Employees: {data.employees}</Item>
                                <Item>Launch Sites: {data.launch_sites}</Item>
                                <Item>Test Sites: {data.test_sites}</Item>
                                <Item>Rockets: {data.vehicles}</Item>
                            </List> : <List></List>
                        } 
                        </TabPanel>
                    </Tabs>
                </Content>
                </Tween>
              </Scroll>
            </Page>
        )
    }
}

const Page = styled.div `
    overflow: hidden;
    height: 1000px;
    width: 100vw;
    font-family: 'Noto Sans', sans-serif;
`;

const Scroll = styled(ScrollOverPack)`
    background: transparent;
    width: 100%;
    height: 1000px;
    padding-top: 10%;
`;

const Content = styled.div`
    margin: 0 auto;
    width: 60%;
    padding: 10px;
    height: 400px;
    background-color: rgba(0,0,0,0.4);
    margin-top: 20px;
    border-radius: 4px;
    color: #ffffff;
`;

const Folder = styled(TabList)`
    display: flex;
    justify-content: flex-start;
    flex-direction: row;
`;

const StyledTab = styled(Tab)`
    list-style: none;
    cursor: pointer;
    color: #8d8d8d;
    width: 100px;
    height: 40px;
    margin-right: 10px;
    border-right: 1px solid orange;
    border-bottom: 1px solid orange;
    border-bottom-right-radius: 8px;
    transition: all .35s ease-in-out;

    &:hover {
        color: white;
        -webkit-box-shadow: 2px -1px 0px -1px orange;
        -moz-box-shadow: 2px -1px 0px -1px orange;
        box-shadow: 2px -1px 0px -1px orange;
        
    }
`;

const Title = styled.h2`
    text-align: center;
`;


const List = styled.ul`
    margin: 80px;
`;

const Item = styled.li`
    list-style: none;
`;

const Item1 = styled.p`
    margin: 80px;
`;

const Tween = styled(TweenOne)`
    opacity: 0;
    transform: translateY(100px);
`;

const Image = styled.div `
    position: absolute;
    top: 4000px;
    left: 0;
    right: 0;
    bottom: 0;
    background-image: url("https://c1.staticflickr.com/5/4888/32040173048_b4a45010a0_m.jpg");
    background-color: #cccccc;
    height: 1000px; 
    background-position: center; 
    background-repeat: no-repeat;
    background-size: cover;
    z-index: -1;
`;