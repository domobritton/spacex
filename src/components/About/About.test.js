import React from 'react'
import About from './index'


jest.mock('../apiGet/apiGet')

it('renders correctly', () => {
    const wrapper = shallow(
        <About />
    )

    expect(wrapper).toMatchSnapshot()
})

it('fetches company info, updates state, and renders on mount', (done) => {
    const wrapper = shallow(<About />)

    setTimeout(() => {
        wrapper.update()

        const state = wrapper.instance().state 
     
        expect(state.company[0].ceo).toEqual('Dom')
        expect(state.company[0].name).toEqual('Front-End Dev')
        expect(state.company[0].founded).toEqual(1974)

        done()
    })
})