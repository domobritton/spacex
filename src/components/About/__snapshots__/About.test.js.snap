// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.img
      src="https://farm5.staticflickr.com/4760/40126462231_c11129efbe_k.jpg"
    />
  </styled.div>
  <Styled(ScrollOverPack)
    id="page3"
    playScale={1}
  >
    <Styled(TweenOne)
      animation={
        Object {
          "opacity": 1,
          "y": 0,
        }
      }
      key="7"
    >
      <styled.div>
        <Tabs
          defaultFocus={false}
          defaultIndex={null}
          forceRenderTabPanel={false}
          selectedIndex={null}
        >
          <Styled(TabList)>
            <Styled(Tab)>
              ABOUT
            </Styled(Tab)>
            <Styled(Tab)>
              INFO
            </Styled(Tab)>
          </Styled(TabList)>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            selectedClassName="react-tabs__tab-panel--selected"
          >
            <styled.h2>
              ABOUT SPACEX
            </styled.h2>
            <styled.p />
          </TabPanel>
          <TabPanel
            className="react-tabs__tab-panel"
            forceRender={false}
            selectedClassName="react-tabs__tab-panel--selected"
          >
            <styled.h2>
              COMPANY INFO
            </styled.h2>
            <styled.ul>
              <styled.li>
                Founded: 
              </styled.li>
              <styled.li>
                Employees: 
              </styled.li>
              <styled.li>
                Launch Sites: 
              </styled.li>
              <styled.li>
                Test Sites: 
              </styled.li>
              <styled.li>
                Rockets: 
              </styled.li>
            </styled.ul>
            <styled.h2>
              COMPANY LEADERSHIP
            </styled.h2>
            <styled.ul>
              <styled.li>
                CEO: 
              </styled.li>
              <styled.li>
                COO: 
              </styled.li>
              <styled.li>
                CTO: 
              </styled.li>
            </styled.ul>
          </TabPanel>
        </Tabs>
      </styled.div>
    </Styled(TweenOne)>
  </Styled(ScrollOverPack)>
</styled.div>
`;
